storage_scrubber/src/scan_pageserver_metadata.rs

```diff
@@ -21,8 +21,8 @@ pub struct MetadataSummary {
     tenant_count: usize,
     timeline_count: usize,
     timeline_shard_count: usize,
-    with_errors: HashSet<TenantShardTimelineId>,
-    with_warnings: HashSet<TenantShardTimelineId>,
+    with_errors: HashMap<String, Vec<String>>,
+    with_warnings: HashMap<String, Vec<String>>,
```

Comment:
Maybe add comments what the strings represent since we removed type information after `TenantShardTimelineId::to_string`.